### PR TITLE
EVG-14182 Add docs to rest subpackages

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: rest/data/doc.go [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
+To add to the Connector, add the method signature into the interface in
+data/data.go. Next, add the implementation that interacts
+with the database to the database backed object. These objects are named by the
+resource they allow access to. The object that allows access to Hosts is called
+DBHostConnector. Finally, add a mock implementation to the mock object. For
+Hosts again, this object would be called MockHostConnector.
+
+Implementing database backed methods requires using methods in Evergreen's model
+package. As much database specific information as possible should be kept out of
+the these methods. For example, if a new aggregation pipeline is needed to complete
```
Comment (Kimchelly) [inline rest/data/doc.go:24]: Remove "the".

File: rest/data/doc.go [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
+the request, it should be defined in the Evergreen model package and used only
```
Comment (Kimchelly) [inline rest/data/doc.go:25]: model package or its relevant subpackage

File: rest/data/doc.go [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
+using a combination of the methods it already contains or when such combination would
+be unseemingly slow or expensive.
 
-	Implementing database backed methods requires using methods in Evergreen's model
-	package. As much database specific information as possible should be kept out of
-	the these methods. For example, if a new aggregation pipeline is needed to complete
-	the request, it should be defined in the Evergreen model package and used only
-	to aggregate in the method.
+To add to the Connector, add the method signature into the interface in
+data/data.go. Next, add the implementation that interacts
```
Comment (Kimchelly) [inline rest/data/doc.go:16]: rest/data/interface.go

File: rest/data/doc.go [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
-	on the main type they allow access to (i.e. all test access is contained in data/test.go).
+Adding to the Connector
 
-	Extending Connector should only be done when the desired functionality cannot be performed
-	using a combination of the methods it already contains or when such combination would
-	be unseemingly slow or expensive.
+The Connector is a very large interface that defines how to access the main
+state of the database and data central to Evergreen's function. All methods of the
+Connector are contained in the data package in files depending
+on the main type they allow access to (i.e. all test access is contained in data/test.go).
```
Comment (Kimchelly) [inline rest/data/doc.go:9]: "test" is super vague as to whether it means "go tests" or "test results" or some other tests, so I would prefer to go with a more obvious example.

(i.e. host access is contained in rest/data/host.go)

File: rest/client/doc.go [hunk 5]
```diff
@@ -0,0 +1,2 @@
+// Package client provides a REST client for the rest package.
```
Comment (Kimchelly) [inline rest/client/doc.go:1]: Can we phrase this instead as "REST client for API requests to the server" or something? It's already obviously in the rest package.

File: rest/route/doc.go [hunk 6]
```diff
@@ -0,0 +1,2 @@
+// Package route handles incoming REST requests for the rest package.
```
Comment (Kimchelly) [inline rest/route/doc.go:1]: Same question as above about "REST requests to the server".

File: rest/model/doc.go [hunk 7]
```diff
@@ -1,35 +1,37 @@
 /*
-	Adding Models
+Package model maps database models to API models for the rest package.
```
Comment (Kimchelly) [inline rest/model/doc.go:2]: Same question as above about "API models for REST requests to the server" (or something along those lines).

**Review by Kimchelly — CHANGES_REQUESTED** [review]

**Review by Kimchelly — APPROVED** [review]

**Comment by bsamek** [comment]
evergreen merge

**Pull request merged**